conans/client/graph/grapher.py

```diff
@@ -1,203 +1,75 @@
-import os
+from jinja2 import Markup
 
 from conans.client.graph.graph import BINARY_BUILD, BINARY_CACHE, BINARY_DOWNLOAD, BINARY_MISSING, \
     BINARY_UPDATE
 from conans.client.installer import build_id
-from conans.util.files import save
 
 
-class ConanGrapher(object):
-    def __init__(self, deps_graph):
-        self._deps_graph = deps_graph
-
-    def graph(self):
-        graph_lines = ['digraph {\n']
-
-        for node in self._deps_graph.nodes:
-            depends = node.neighbors()
-            if depends:
-                depends = " ".join('"%s"' % str(d.ref) for d in depends)
-                graph_lines.append('    "%s" -> {%s}\n' % (node.conanfile.display_name, depends))
-
-        graph_lines.append('}\n')
-        return ''.join(graph_lines)
-
-    def graph_file(self, output_filename):
-        save(output_filename, self.graph())
+class Node(object):
+    def __init__(self, id, node, is_build_time_node):
+        self.id = id
+        self._ref = node.ref
+        self._conanfile = node.conanfile
+        self._is_build_time_node = is_build_time_node
+        self.package_id = node.package_id
+        self.binary = node.binary
+
+    @property
+    def label(self):
+        return self._ref or self._conanfile.display_name
+
+    @property
+    def is_build_requires(self):
+        return self._is_build_time_node
+
+    def data(self):
+        def format_url(url):
+            return Markup('<a href="{url}">{url}</a>'.format(url=url))
+
+        def join_if_iterable(value):
+            if isinstance(value, (list, tuple)):
+                return ", ".join(value)
+            return value
+
+        return {
+            'build_id': build_id(self._conanfile),
+            'url': format_url(self._conanfile.url),
+            'homepage': format_url(self._conanfile.homepage),
+            'license': self._conanfile.license,
+            'author': self._conanfile.author,
+            'topics': join_if_iterable(self._conanfile.topics)
+        }
 
 
-class ConanHTMLGrapher(object):
-    def __init__(self, deps_graph, cache_folder):
+class Grapher(object):
+    def __init__(self, deps_graph):
         self._deps_graph = deps_graph
-        self._cache_folder = cache_folder
+        self.nodes, self.edges = self._build_graph()
 
-    def _visjs_paths(self):
```

Comment:
please do not remove this feature, it is the only way users without internet access can use the html graph.